deepchecks/vision/checks/performance/robustness_report.py

```diff
@@ -0,0 +1,244 @@
+
+        fig = (
+            fig.update_xaxes(title=None, type='category')
+                .update_yaxes(title=None, matches=None)
+                .for_each_annotation(lambda a: a.update(text=a.text.split('=')[-1]))
+                .for_each_yaxis(lambda yaxis: yaxis.update(showticklabels=True))
+        )
+
+        return CheckResult(
+            results_df,
+            header='Robustness Report',
+            display=fig
+        )
+
+    def _report_inner_loop(self, dataset: VisionDataset, model, scorers):
+        n_samples = dataset.get_samples_per_class()
+        classes = n_samples.keys()
+
+        def evaluate(data_loader):
+            return (
+                [class_name, name, class_score, n_samples[class_name]]
+                for name, score in calculate_metrics(list(scorers.values()), dataset, data_loader, model,
+                                                     prediction_extract=self.prediction_extract).items()
+                # scorer returns numpy array of results with item per class
+                for class_score, class_name in zip(score, classes)
+            )
+
+        # Run baseline
+        curr_results_base = evaluate(dataset.get_data_loader())
+        results_base_df = pd.DataFrame(curr_results_base, columns=['Class', 'Metric', 'Value', 'N']
+                                       ).sort_values(by=['Class'])
+
+        results = []
+        examples = []
+        for curr_aug in self.augmentations:
+            aug_name = augmentation_name(curr_aug)
+            # Create a copy of data loader and the dataset
+            copy_data_loader = copy(dataset.get_data_loader())
+            copy_data_loader.dataset = copy(copy_data_loader.dataset)
+            # Add augmentation in the first place
+            add_augmentation(copy_data_loader.dataset, dataset.transform_field, curr_aug)
+            # Evaluate
+            curr_results_aug = evaluate(copy_data_loader)
+            results_aug_df = pd.DataFrame(curr_results_aug, columns=['Class', 'Metric', 'Value', 'N']
+                                          ).sort_values(by=['Class'])
+            # This modifies the augmentation name in-place
+            results_base_df["Augmentation"] = aug_name
+            # This adds the augmentation
+            results_aug_df["Augmentation"] = aug_name
+            results.append(pd.concat([results_base_df, results_aug_df], keys=["Baseline", "Augmented"]))
+
+            examples.append(self._get_random_image_pairs_from_dataloder(dataset.get_data_loader(), copy_data_loader))
+
+        # Create grand DataFrame from dictionary of augmentations
+        # This has flat hierarchy for augmentations and baseline-vs-augmentations
+        results_df = pd.concat(results, axis=0). \
+            reset_index(). \
+            rename({"level_0": "Status"}, axis=1). \
+            drop(labels="level_1", axis=1)
+
+        def _is_robust_to_augmentation(p, epsilon=10 ** -2):
```

Comment:
should be a condition.

Reply:
We have a problem with the condition, how do we decide what augs to show to the user without this?